pallets/creator-staking/src/types/mod.rs

```diff
@@ -0,0 +1,202 @@
+/// 2. In era **6**, staker staked additional **500**, increasing total staked amount to **1500**
+/// 3. No entry for era **7** exists which means there were no changes from the former entry.
+///    This means that in era **7**, staked amount was also **1500**
+/// 4. In era **8**, staker staked an additional **600**, increasing total stake to **2100**
+/// 5. In era **9**, staker unstaked everything from the creator (interpreted from `<9, 0>`)
+/// 6. No changes were made in era **10** so we can interpret this same as the previous entry which
+/// means **0** staked amount. 7. In era **11**, staker staked **500** on the creator, making his
+/// stake active again after 2 eras of inactivity.
+///
+/// **NOTE:** It is important to understand that staker **DID NOT** claim any rewards during this
+/// period.
+#[derive(Encode, Decode, Clone, Default, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct StakerInfo<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> {
+    // Size of this list would be limited by a configurable constant
+    stakes: Vec<EraStake<Balance>>,
+}
+
+/// Represents an balance amount undergoing the unbonding process.
+/// Since unbonding takes time, it's important to keep track of when and how much was unbonded.
+#[derive(
+Clone, Copy, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo, MaxEncodedLen,
+)]
+pub struct UnlockingChunk<Balance: MaxEncodedLen> {
+    /// Amount being unlocked
+    #[codec(compact)]
+    pub(super) amount: Balance,
+    /// Era in which the amount will become unlocked and can be withdrawn.
+    #[codec(compact)]
+    pub(super) unlock_era: EraIndex,
+}
+
+/// Contains unlocking chunks.
+/// This is a convenience struct that provides various utility methods to help with unbonding
+/// handling.
+#[derive(Clone, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+pub struct UnbondingInfo<Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen> {
+    // Vector of unlocking chunks. Sorted in ascending order in respect to unlock_era.
+    unlocking_chunks: Vec<UnlockingChunk<Balance>>,
+}
+
+/// A record of rewards allocated for stakers and creators
+#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct RewardInfo<Balance: HasCompact + MaxEncodedLen> {
+    /// Total amount of rewards for stakers in an era
+    #[codec(compact)]
+    pub stakers: Balance,
+    /// Total amount of rewards for creators in an era
+    #[codec(compact)]
+    pub creators: Balance,
+}
+
+/// A record for total rewards and total amount staked for an era
+#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EraInfo<Balance: HasCompact + MaxEncodedLen> {
+    /// Total amount of earned rewards for an era
+    pub rewards: RewardInfo<Balance>,
+    /// Total staked amount in an era
+    #[codec(compact)]
+    pub staked: Balance,
+    /// Total locked amount in an era
```

Comment:
Need to explain, what is the difference b/w staked and locked. Does locked include staked? Etc.

Reply:
Locked include staking, but when you unstake, during unbonding period, there is some amount in locked, which is not present in staked.

Reply:
```suggestion
    /// Total locked amount in an era: stake locks + unbonding locks.
```